source/brailleDisplayDrivers/hidBrailleStandard.py

```diff
@@ -226,13 +239,20 @@ def display(self, cells: List[int]):
 		# cells will already be padded up to numCells.
 		padded_cells = cells + [0] * (self._maxNumberOfCells - len(cells))
 		cellBytes = b"".join(intToByte(cell) for cell in padded_cells)
-		report = hwIo.hid.HidOutputReport(self._dev, reportID=self._cellValueCaps.ReportID)
```

Comment:
It took me quite a while to fully follow this method.
I think this can be greatly simplified if you use self.numRows and self.numCols and do some classic 2d array math: 
E.g. 
```
		for rowIndex in range(self.numRows):
			startOffset = rowIndex * self.numCols
			endOffset = startOffset + self.numCols
			reportID = self._cellValueCaps[rowIndex].ReportID
			report = hwIo.hid.HidOutputReport(self._dev, reportID=reportID)
			report.setUsageValueArray(
				HID_USAGE_PAGE_BRAILLE,
				valueCap.LinkCollection,
				valueCap.u1.NotRange.Usage,
				cellBytes[startOffset: endOffset]
			)
			self._dev.write(report.data)
```
Though I have not tested this code.

Reply:
@alexmoon  fair enough. Thanks for the clear reasoning.
I shall approve as is. However, can you provide some further argument as to why a device would choose to place multiple rows in the one report? Is it just to bulk write calls? I'm wondering that if in the long-term if we were to improve NVDA's support for multiline such that you could change an individual row without updating the entire display, would this be possible if a report contained more than one row?

Reply:
@michaelDCurran Would it make some sense to have additional explanatory comments in the code about what's going on here? If even you had a trial trying to determine why this had to be written this way, it's reasonable to think some more comments might help future devs.

Reply:
> @alexmoon fair enough. Thanks for the clear reasoning. I shall approve as is. However, can you provide some further argument as to why a device would choose to place multiple rows in the one report? Is it just to bulk write calls?

Yes, each report written has to be in a separate USB or Bluetooth packet. If the rows are short, combining them into a single report can significantly improve latency, especially over Bluetooth.

> I'm wondering that if in the long-term if we were to improve NVDA's support for multiline such that you could change an individual row without updating the entire display, would this be possible if a report contained more than one row?

The driver would need to cache the data for the other rows so that it could build the full report.

> @michaelDCurran Would it make some sense to have additional explanatory comments in the code about what's going on here? If even you had a trial trying to determine why this had to be written this way, it's reasonable to think some more comments might help future devs.

Yes, I can add some additional comments.

Reply:
@michaelDCurran I've reworked the display method and added some additional comments. Hopefully it is easier to follow now.